eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxOut.hs

```diff
@@ -226,26 +223,28 @@ encodeAddress28 network paymentCred = do
           ScriptHashObj {} -> 0
       encodeAddr ::
         Hash ADDRHASH a ->
-        Maybe (SizeHash ADDRHASH :~: 28, Addr28Extra)
+        Addr28Extra
       encodeAddr h = do
-        refl@Refl <- sameNat (Proxy @(SizeHash ADDRHASH)) (Proxy @28)
         case hashToPackedBytes h of
           PackedBytes28 a b c d ->
             let d' = (fromIntegral d `shiftL` 32) .|. networkBit .|. payCredTypeBit
-             in Just (refl, Addr28Extra a b c d')
-          _ -> Nothing
+             in Addr28Extra a b c d'
+          _ ->
+            -- This case should never match, but if it does, we output garbage
+            Addr28Extra 0xaa 0xaa 0xaa 0xaa
```

Comment:
Please, never do hacks like this. It is better to throw an `error "Impossible"` then make up bogus values.
```suggestion
```

In any case, this case is not needed, since `Maybe` is out of the picture

Reply:
There's the [`PackedBytes#`](https://github.com/IntersectMBO/cardano-base/blob/master/cardano-crypto-class/src/Cardano/Crypto/PackedBytes.hs#L71) constructor which matches any `n`, so the typechecker will complain about that case not being covered. That case should never happen, because the size is known statically and the constructing function will pick the correct constructor, but I still have to cover it somehow to make the typechecker happy.